designdocs/recovery_system.md

```diff
@@ -0,0 +1,609 @@
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum AuthenticationContactType {
+    /// Email address contact type
+    Email = 0x00,
+    /// Phone number contact type (E.164 format)
+    PhoneNumber = 0x01,
+    /// Future extension for other contact types
+    /// Could include social media handles, attestation-based contacts, etc.
+}
+
+/// Authentication Contact structure with type-safe encoding
+/// Format: [type_byte] + [contact_data_bytes]
+/// Example: 0x00 + utf8_to_bytes("user@example.com")
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct AuthenticationContact {
+    pub contact_type: AuthenticationContactType,
+    pub contact_data: Vec<u8>,
+}
+
+/// New Events for the Recovery System
+/// These events are emitted by the MSA pallet to indicate successful operations related to Recovery Secrets.
+#[pallet::event]
+#[pallet::generate_deposit(pub(super) fn deposit_event)]
+pub enum Event<T: Config> {
+    /// A Recovery Commitment was added to an existing MSA
+    RecoveryCommitmentAdded {
+        who: T::AccountId,
+        msa_id: MsaId,
+    },
+    /// Recovery Intermediary Hashes were verified successfully
+    RecoveryCommitmentVerified {
+        who: T::AccountId,
+        msa_id: MsaId,
+    },
+    /// An account was recovered with a new control key
+    AccountRecovered {
+        who: T::AccountId,
+        msa_id: MsaId,
+        new_control_key: T::PublicKey, // New control key added to the MSA
+    },
+    /// A Recovery Commitment was invalidated after use or removal
+    RecoveryCommitmentInvalidated {
+        who: T::AccountId,
+        msa_id: MsaId,
+    },
+}
+```
+
+### Storage Structures
+
+```rust
+/// Storage type for a MsaID to Recovery Commitment information
+/// - Key: MsaId
+/// - Value: RecoveryCommitment
+#[pallet::storage]
+pub type MsaIdToRecoveryCommitment<T: Config> = StorageMap<_, Twox64Concat, MsaId, RecoveryCommitment>;
+
+// TODO: Explore efficiency for this storage and if we can avoid double indexing
+/// Storage type for a Recovery Commitment to MsaId mapping
+/// - Key: RecoveryCommitment
+/// - Value: MsaId
+#[pallet::storage]
+pub type RecoveryCommitmentToMsaId<T: Config> = StorageMap<_, Twox64Concat, RecoveryCommitment, MsaId>;
+```
+
+## Required Extrinsics
+
+|Name/Description|Caller|Payment|Key Events|Runtime Added|
+|---|---|---|---|---|
+| `add_recovery_commitment`<br />Add a new Recovery Commitment to an existing MSA| Provider| Capacity or Tokens | [`RecoveryCommitmentAdded`](https://frequency-chain.github.io/frequency/pallet_msa/pallet/enum.Event.html#variant.RecoveryCommitmentAdded)| 166|
+| `verify_intermediary_hashes`<br />Verify Recovery Intermediary Hashes against stored Recovery Commitment| Recovery Provider| Capacity or Tokens | [`RecoveryCommitmentVerified`](https://frequency-chain.github.io/frequency/pallet_msa/pallet/enum.Event.html#variant.RecoveryCommitmentVerified)| 166|
```

Comment:
This wouldn't be an Extrinsic, just an internal function or perhaps a Runtime RPC right?